### PR TITLE
Feature/449934 trigger azure function implement orchestrator

### PR DESCRIPTION
**Pull request by Skumarhpa**

### CODE REVIEW

File: src/EPR.Calculator.Service.Function/Constants/Config.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace EPR.Calculator.Service.Function.Constants
+{
+    public class Config
+    {
+        public const string PipelineUrl = "PipelineUrl";
```
Comment (johnbenjamin-defra) [inline src/EPR.Calculator.Service.Function/Constants/Config.cs:5]: Are we not creating constants as discussed?

File: src/EPR.Calculator.Service.Common/CalculatorRunConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (udaydefra) [inline src/EPR.Calculator.Service.Common/CalculatorRunConfiguration.cs:2]: Remove any unused usings

File: src/EPR.Calculator.Service.Function/Interface/ICalculatorRunService.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EPR.Calculator.Service.Function.Interface
+{
+    public interface ICalculatorRunService
+    {
+        public void startProcess(CalculatorRunParameter calculatorRunParameter);
```
Comment (udaydefra) [inline src/EPR.Calculator.Service.Function/Interface/ICalculatorRunService.cs:12]: Function name should start with Capital

File: src/EPR.Calculator.Service.Function/Services/CalculatorRunService.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+
+namespace EPR.Calculator.Service.Function.Services
+{
+    public class CalculatorRunService : ICalculatorRunService
+    {
+             
+        public void startProcess(CalculatorRunParameter calculatorRunParameter)
+        {
+           var environmentConfiguration =   GetConfiguration(true);
```
Comment (udaydefra) [inline src/EPR.Calculator.Service.Function/Services/CalculatorRunService.cs:14]: Why are we using true as a parameter? Do we need a variable for it?

File: src/EPR.Calculator.Service.Function/ServiceBusQueueTrigger.cs [hunk 5]
```diff
@@ -1,15 +1,40 @@
...
         [FunctionName("EPRCalculatorRunServiceBusQueueTrigger")]
-        public static void Run([ServiceBusTrigger(queueName: "%ServiceBusQueueName%", Connection = "ServiceBusConnectionString")] string myQueueItem, ILogger log)
+        public IActionResult Run([ServiceBusTrigger(queueName: "%ServiceBusQueueName%", Connection = "ServiceBusConnectionString")] string myQueueItem, ILogger log)
         {
-            log.LogInformation($"C# Service Bus Queue trigger function proceed messages: {myQueueItem}");
 
+            if (string.IsNullOrEmpty(myQueueItem)) { return new BadRequestResult(); }
+
+            var calculatorRunParameter = CalculatorRunParameterMapper.Map(JsonConvert.DeserializeObject<CalculatorParameter>(myQueueItem));
+            _calculatorRunService.startProcess(calculatorRunParameter);
```
Comment (udaydefra) [inline src/EPR.Calculator.Service.Function/ServiceBusQueueTrigger.cs:35]: Do we need to validate incoming CalculatorRunParameter?

File: src/EPR.Calculator.Service.Common/CalculatorRunConfiguration.cs (hunk 2 above)
Comment (Skumarhpa) [reply src/EPR.Calculator.Service.Common/CalculatorRunConfiguration.cs]: Resolved the comment.

File: src/EPR.Calculator.Service.Function/Interface/ICalculatorRunService.cs (hunk 3 above)
Comment (Skumarhpa) [reply src/EPR.Calculator.Service.Function/Interface/ICalculatorRunService.cs]: Updated the Function name

File: src/EPR.Calculator.Service.Function/Services/CalculatorRunService.cs (hunk 4 above)
Comment (Skumarhpa) [reply src/EPR.Calculator.Service.Function/Services/CalculatorRunService.cs]: created a variable for the same.

File: src/EPR.Calculator.Service.Function/ServiceBusQueueTrigger.cs (hunk 5 above)
Comment (Skumarhpa) [reply src/EPR.Calculator.Service.Function/ServiceBusQueueTrigger.cs]: included the try catch blocks to handle the issues.

File: src/EPR.Calculator.Service.Function/ServiceBusQueueTrigger.cs [hunk 6]
```diff
@@ -1,15 +1,51 @@
...
+
         [FunctionName("EPRCalculatorRunServiceBusQueueTrigger")]
-        public static void Run([ServiceBusTrigger(queueName: "%ServiceBusQueueName%", Connection = "ServiceBusConnectionString")] string myQueueItem, ILogger log)
+        public void Run([ServiceBusTrigger(queueName: "%ServiceBusQueueName%", Connection = "ServiceBusConnectionString")] string myQueueItem, ILogger log)
         {
-            log.LogInformation($"C# Service Bus Queue trigger function proceed messages: {myQueueItem}");
 
+            if (string.IsNullOrEmpty(myQueueItem))
+            {
+                log.LogError($"Message is Null or empty");
```
Comment (udaydefra) [inline src/EPR.Calculator.Service.Function/ServiceBusQueueTrigger.cs:32]: Should this return?

Comment (Skumarhpa) [reply src/EPR.Calculator.Service.Function/ServiceBusQueueTrigger.cs]: resolved the comment.Thanks

**Review by udaydefra — APPROVED** [review]

**Review by johnbenjamin-defra — APPROVED** [review]

**Pull request merged**